src/services/selectorPrinting.ts

```diff
@@ -470,6 +469,27 @@ export class SelectorPrinting {
 							continue elementLoop;
 						}
 
+						if (text.match(/^:(?:nth-child|nth-last-child)/i) && childElements.length > 0) {
+							/* The specificity of the :nth-child(An+B [of S]?) pseudo-class is the specificity of a single pseudo-class plus, if S is specified, the specificity of the most specific complex selector in S */
+							// https://www.w3.org/TR/selectors-4/#the-nth-child-pseudo
+							specificity.attr++;
+
+							const selectorText = text.slice(text.indexOf('(') + 1, text.length -1);
+
+							// Test for presence of complex-selector-list S; find highest specificity
+							const keyword = text.indexOf(' of ');
```

Comment:
Testing for a space before and after `of` is not clean. There could also be a tab, or even a non word character
```
ul.one > li:nth-child(-n + 3 of.noted) {
}
```
would be valid too.

I suggest to jeust use the scanner to find out if there's an `of` ident in the string.

Reply:
Reworked the logic in the latest commit. When the 'An + B [of S]?' syntax is respected then we can just grab the last childElement as it will contain the entire complex-selector list (S). 

The edge case where 'n' has no integer prefix (the parser does not recognise it as a binary expression token). Extract out the complex-selector list, and re-parse the selectors. This also handles the different syntax possibilities; added more tests for completeness.